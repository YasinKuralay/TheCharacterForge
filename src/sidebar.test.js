import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Sidebar, { keyCheck, inputKeyCheck } from "./sidebar.js";

//Mocking the char prop of Sidebar
let charInfo = { name: "Some random placeholder name" };

it("renders the Sidebar component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Sidebar char={charInfo} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("returns nothing if the key pressed is not Enter", () => {
    expect(keyCheck("a")).toBeUndefined();
    expect(keyCheck("A")).toBeUndefined();
    expect(keyCheck("k")).toBeUndefined();
    expect(keyCheck("l")).toBeUndefined();
});

it("returns nothing if the key pressed is not Enter", () => {
    expect(inputKeyCheck("T")).toBeUndefined();
    expect(inputKeyCheck("d")).toBeUndefined();
    expect(inputKeyCheck("x")).toBeUndefined();
    expect(inputKeyCheck("i")).toBeUndefined();
});

it("matches the previous snapshot of the Sidebar Component", () => {
    const SidebarSnapshot = renderer
        .create(<Sidebar char={charInfo} />)
        .toJSON();
    expect(SidebarSnapshot).toMatchSnapshot();
});
