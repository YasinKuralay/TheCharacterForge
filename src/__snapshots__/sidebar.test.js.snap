// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot of the Sidebar Component 1`] = `
<div
  id="sidebar"
>
  <div
    className="sidebarCard sidebarPersonName"
  >
    <i
      className="fas fa-portrait"
    />
     
    <input
      autoComplete="off"
      defaultValue="Some random placeholder name"
      id="charNameField"
      onKeyDown={[Function]}
    />
  </div>
  <div
    id="addBar"
  >
    <i
      className="fas fa-folder-plus"
    />
    <i
      className="fas fa-plus-square"
      onClick={[Function]}
    />
  </div>
</div>
`;
